### PR TITLE
response table added

### PR DESCRIPTION
**Pull request by lrocheWB**

### CODE REVIEW

**Comment by kinnou02** [comment]
You need to create a migration for creating the "response" table.

By the way I don't really like the name, since it don't really contain the response, but I honestly don't have any better idea.

**Comment by lrocheWB** [comment]
Indeed, I have forgot to ask for create table but I knew it ;-) 

For the name, we will see...

File: stat_persistor/saver/stat_request.py [hunk 1]
```diff
@@ -124,10 +131,20 @@ def build_stat_request_dict(stat):
...
+    return{
+        'object_count': response.object_count,
+        'request_id': request_id
+    }
+
 def build_journey_request_dict(journey_request, request_id):
     """
     Construit à partir d'un object protobuf pbnavitia.stat.HitStat
-    Utilisé pour l'insertion dans la table stat.requests
+    Utilisé pour l'insertion dans la table stat.journey_request
```
Comment (antoine-de) [inline stat_persistor/saver/stat_request.py]: I know it's not you doing, but I don't know how those comments came to be accepted in the repo, all comments must be in english :wink:

Comment (lrocheWB) [inline stat_persistor/saver/stat_request.py]: Someone, in the past, have doing a bad copy paste. I'm agree, all comments must be in english...

File: stat_persistor/saver/stat_request.py [hunk 2]
```diff
@@ -50,6 +50,7 @@ def persist_stat_request(meta, conn, stat):
     interpreted_parameters_table = meta.tables['stat.interpreted_parameters']
     journey_request_table = meta.tables['stat.journey_request']
     filter_table = meta.tables['stat.filter']
+    response_table = meta.tables['stat.response']
```
Comment (pbougue) [inline stat_persistor/saver/stat_request.py]: For the name, I think it's the same discuss than https://github.com/CanalTP/navitia-proto/pull/14/files#r41487800

Comment (lrocheWB) [inline stat_persistor/saver/stat_request.py]: Exactly :)

File: migrations/alembic/versions/fos5a55d7a_add_response.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
+"""add journey_request
```
Comment (kinnou02) [inline migrations/alembic/versions/fos5a55d7a_add_response.py]: This comment is shown when the migration is executed, so it's important to describe what it's done in this migration.

Comment (lrocheWB) [inline migrations/alembic/versions/fos5a55d7a_add_response.py]: Thx. Bad copy paste.

File: stat_persistor/saver/stat_request.py [hunk 4]
```diff
@@ -124,10 +131,20 @@ def build_stat_request_dict(stat):
         'end_point_name': stat.end_point_name,
     }
 
+def build_stat_response_dict(response, request_id):
+    """
+    Build from protobuf object pbnavitia.stat.HitStat
+    Use to insert in stat.response table
+    """
+    return{
```
Comment (xlqian) [inline stat_persistor/saver/stat_request.py]: space between return and {

File: stat_persistor/saver/stat_request.py [hunk 5]
```diff
@@ -124,10 +131,20 @@ def build_stat_request_dict(stat):
...
+        'item_on_page': response.item_on_page,
+        'request_id': request_id
+    }
+
 def build_journey_request_dict(journey_request, request_id):
     """
-    Construit à partir d'un object protobuf pbnavitia.stat.HitStat
-    Utilisé pour l'insertion dans la table stat.requests
+    Build from protobuf object pbnavitia.stat.HitStat
```
Comment (pbougue) [inline stat_persistor/saver/stat_request.py]: :+1:

File: migrations/alembic/versions/fos5a55d7a_add_response.py [hunk 6]
```diff
@@ -0,0 +1,26 @@
+"""add response
```
Comment (is06) [inline migrations/alembic/versions/fos5a55d7a_add_response.py]: Change this to info_response please

File: migrations/alembic/versions/fos5a55d7a_add_response.py [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+revision = 'fos5a55d7a'
+down_revision = '14a0ba4227ce'
+
+from alembic import op
+import sqlalchemy as sa
+
+def upgrade():
+    op.create_table(
+        'response',
+        sa.Column('item_on_page', sa.Integer(), nullable=False),
```
Comment (is06) [inline migrations/alembic/versions/fos5a55d7a_add_response.py]: Change this to object_count

File: migrations/alembic/versions/fos5a55d7a_add_response.py [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+        sa.Column('request_id', sa.BigInteger(), nullable=False),
+        sa.ForeignKeyConstraint(['request_id'], ['stat.requests.id'],),
+        schema='stat'
+    )
+
+def downgrade():
+    op.drop_table('response', schema='stat')
```
Comment (is06) [inline migrations/alembic/versions/fos5a55d7a_add_response.py]: Change response to info_response

File: stat_persistor/saver/stat_request.py (hunk 2 above)
Comment (is06) [inline stat_persistor/saver/stat_request.py]: Change response_table to info_response_table and stat.response to stat.info_response

File: stat_persistor/saver/stat_request.py [hunk 9]
```diff
@@ -124,10 +131,20 @@ def build_stat_request_dict(stat):
         'end_point_name': stat.end_point_name,
     }
 
+def build_stat_response_dict(response, request_id):
+    """
+    Build from protobuf object pbnavitia.stat.HitStat
+    Use to insert in stat.response table
+    """
+    return{
+        'item_on_page': response.item_on_page,
```
Comment (is06) [inline stat_persistor/saver/stat_request.py]: Change response to info_response
And item_on_page to object_count

File: stat_persistor/saver/stat_request.py (hunk 5 above)
Comment (pbougue) [inline stat_persistor/saver/stat_request.py]: I'd translate it by `Built` above and `Used` below, though

**Comment by pbougue** [comment]
Isn't there still that size of db problem?
I thought you didn't want to persist every count for every response in db, at least in prod (maybe just accumulate by user or so).
But if it's the goal and you've checked it with @vincentlepot that PR is ok for me.

**Comment by vincentlepot** [comment]
We still have the size issue in production. For the moment, we should not deploy this stat-persistor in production, but only up to customer.
I'll check with exploitation for the db size

**Comment by pbougue** [comment]
@lrocheWB : Ok, so should I merge that PR (and let you deal where you deploy) or do you want to change it (to lower db size and even avoid questions on prod space) ?

**Comment by lrocheWB** [comment]
@pbougue: ok you can merge that PR :)

File: stat_persistor/saver/stat_request.py (hunk 5 above)
Comment (pbougue) [inline stat_persistor/saver/stat_request.py]: This change will be for next PR...

**Pull request merged**